src/orion/core/worker/multi_task_algo.py

```diff
@@ -0,0 +1,551 @@
+def _(algo: str) -> bool:
+    available_algos = get_all_algos()
+    if algo not in available_algos and algo.lower() not in available_algos:
+        raise RuntimeError(
+            f"Can't tell if algo '{algo}'' is warm-starteable, since there are no "
+            f"algos registered with that name!\n"
+            f"Available algos: {list(available_algos.keys())}"
+        )
+    algo_type = available_algos.get(algo, available_algos.get(algo.lower()))
+    assert algo_type is not None
+    return inspect.isclass(algo_type) and issubclass(algo_type, WarmStarteable)
+
+
+@is_warmstarteable.register(dict)
+def _(algo: Dict[str, Any]) -> bool:
+    first_key = list(algo)[0]
+    return len(algo) == 1 and is_warmstarteable(first_key)
+
+
+class MultiTaskAlgo(AlgoWrapper):
+    """Wrapper that makes the algo "multi-task" by concatenating the task ids to the inputs.
+    """
+
+    def __init__(
+        self,
+        space: Space,
+        algorithm_config: Dict,
+        knowledge_base: AbstractKnowledgeBase,
+    ):
+        log.debug(
+            f"Creating a MultiTaskAlgo wrapper: space {space}, algo config: {algorithm_config}"
+        )
+        self.knowledge_base = knowledge_base
+        self._enabled: bool
+        if is_warmstarteable(algorithm_config):
+            log.debug(
+                "Chosen algorithm is warm-starteable, disabling this multi-task wrapper."
+            )
+            self.algorithm = PrimaryAlgo(space=space, algorithm_config=algorithm_config)
+            self.space = self.algorithm.space
+            self._enabled = False
+            return  # Return, just to indicate we don't do the other modifications below.
+        else:
+            log.debug(
+                "Chosen algorithm is NOT warm-starteable, enabling this multi-task wrapper."
+            )
+            self._enabled = True
+
+        # Figure out the number of potential tasks using the KB.
+        # TODO: The current Experiment might already be registered in the KB?
+        if knowledge_base.n_stored_experiments < 1:
+            # TODO: Should probably turn the 'warm-start' portion of this 'wrapper' off,
+            # because it will probably reduce the effectiveness of the algo to have to
+            # optimize with an uninformative 'task_id' dimension.
+            warnings.warn(
+                RuntimeWarning(
+                    f"Making the Algo 'multi-task', even though there are only"
+                    f"{knowledge_base.n_stored_experiments} experiments in the "
+                    f"Knowledge base. This might reduce the efficiency of the algo."
+                )
+            )
+        # TODO: Should we have this dimension be larger than the current number of
+        # experiments in the KB, in case some get added in the future?
+        max_task_id = self.knowledge_base.n_stored_experiments + 1
+        task_label_dimension = Categorical(
+            "task_id", list(range(0, max_task_id)), default_value=0,
```

Comment:
This implies that no hyperparameter may be named `task_id`. I think it's fine but it should be clearly stated in the main docstring of the class.